query/common/Util.java

```diff
@@ -31,8 +31,8 @@ public static RoleType tryInferRoleType(Relation relation, Thing player, Relatio
             if ((roleType = relationType.getRelates(var.label().get().label())) == null) {
                 throw TypeDBException.of(TYPE_NOT_FOUND, Label.of(var.label().get().label(), relationType.getLabel().name()));
             }
-        } else if ((inferred = player.getType().getPlays()
-                .filter(rt -> rt.getRelationType().equals(relation.getType()))
+        } else if ((inferred = relation.getType().getRelates()
+                .filter(relatesRole -> player.getType().plays(relatesRole))
```

Comment:
Had to flip this, because we have `ThingType.plays(RoleType) -> boolean`, but not `RelationType.relates(RoleType) -> boolean`